File: server/BeerWorkshop.Application/Dto/Responses/ProductsInventory/ProductInventoryItemResponseDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Responses.ProductsInventory;

public record ProductInventoryItemResponseDto(
    Guid Id,
    decimal Quantity,
    DateTime IncomingDate,
    DateTime? OpeningDate,
    decimal PurchasePrice,
    decimal RetailPrice,
    decimal PricePerQuantity,
    DateTime ManufactureDate,
    int ExpirationTime,
    ExpirationMeasure ExpirationMeasure,
    ExpirationCountingDateType ExpirationCountingDateType
);

File: client/src/redux/dtos/responses/productsInventory/productInventoryItemResponseDto.ts
import {
  ExpirationCountingDateType,
  ExpirationMeasureType,
} from "../../../enums";

export interface ProductInventoryItemResponseDto {
  id: string;
  quantity: number;
  incomingDate: Date;
  purchasePrice: number;
  purchaseVat: number;
  retailPrice: number;
  pricePerQuantity: number;
  manufactureDate: Date;
  expirationTime: number;
  expirationMeasure: ExpirationMeasureType;
  openingDate?: Date;
  expirationCountingDateType: ExpirationCountingDateType;
}
